### PR TITLE
3d Pie Chart and pop up modal component -> 1. Alerts Vs National park count  2. Alerts Vs. park count

### PR DESCRIPTION
**Pull request by rutujapadgilwar**
## Description:
This pull request introduces two pie charts: one for comparing the counts of national parks per alerts category and another for comparing the overall park counts per alerts category. The park and alerts were endpoints utilized to gather the necessary data for these charts.
Moreover, A modal component has been implemented, which is triggered when a user clicks on a section in the pie chart.

## Changes Made: 
1. I have implemented a single Alerts3dPieChart component that has been created to process the data for the pie chart. This component is reused in both charts to avoid code duplication and promote code reusability.
2. A modal component has been implemented, which is triggered when a user clicks on a column in the bar chart. The modal displays a list of park names for the selected alert category, along with the corresponding park images. Each park name is a link that redirects the user to the "parkInfoPage".
3. To showcase the park images and park names in the modal, I have integrated the park data with the alert data to ensure precise and reliable outcomes.
4. The modal view has been updated to display two parks per row if a state has more than 10 parks, otherwise it shows one park per row.

## Screenshots:
Alerts Vs all park count
image

Alerts Vs National Parks
image

Modal view:
image

### CODE REVIEW

File: park-dashboard/src/Pages/NationalParkCharts/NationalParkChartsPage.js [hunk 1]
```diff
@@ -3,27 +3,36 @@ import { Container, Row, Col } from "react-bootstrap";
...
+      <Container className="barchart" centered style={{ marginBottom: "4rem" }}>
         <Row>
           <h2 style={{ fontSize: "18px" }} className="text-center">
             Bar Chart: State Vs Park Counts
           </h2>
           <StateVsNationalParks />
         </Row>
       </Container>
-      <Container className="donut-chart" centered>
+      <Container>
```
Comment (juhiaugustine) [inline park-dashboard/src/Pages/NationalParkCharts/NationalParkChartsPage.js:21]: I remember adding some code to make the charts more responsive? Can you pull the latest main branch and then modify your code here? 

This is my PR where I added code to make the charts responsive -> https://github.com/rutujapadgilwar/Park-Dashboard/pull/13/files#diff-110bc809b1b51a37356fc9ef0586e5e43f81888620da72233e2232c0a8e90355

You can probably replace the second col with your component.

image

File: park-dashboard/src/Pages/Generic Charts/GenericCharts.js [hunk 2]
```diff
@@ -35,6 +35,14 @@ const GenericChart = () => {
           </Col>
         </Row>
       </Container>
+      <Container style={{ marginBottom: "5rem", marginTop: "12rem"}}>
+        <Col>
+          <h2 style={{ fontSize: "18px" }} className="text-center">
+            3D Pie Chart: Alerts Vs Park Counts
+          </h2>
+          <AlertsVsParks />
```
Comment (juhiaugustine) [inline park-dashboard/src/Pages/Generic Charts/GenericCharts.js:43]: I am seeing a crash when I tap on `Park Closure`. Can you fix it?

Steps to reproduce:

- Go to Generic charts page
- Scroll down to `AlertsVsParks` component
- Tap on `Park Closure` 

image

File: park-dashboard/src/Pages/NationalParkCharts/NationalParkChartsPage.js [hunk 3]
```diff
@@ -3,27 +3,36 @@ import { Container, Row, Col } from "react-bootstrap";
...
+              Donut Chart: Activities vs National Parks
+            </h2>
+            <ActivitiesVsNP className="py-5 activities" />
+          </Col>
         </Row>
+        <Col>
+          <h2 style={{ fontSize: "18px" }} className="text-center">
+            3D Pie Chart: Alerts Vs National Park Counts
+          </h2>
+          <AlertsVsNationalParks className="alerts" />
```
Comment (juhiaugustine) [inline park-dashboard/src/Pages/NationalParkCharts/NationalParkChartsPage.js:34]: I remember adding some code to make the charts more responsive? Can you pull the latest main branch and then modify your code here? 

This is my PR where I added code to make the charts responsive -> https://github.com/rutujapadgilwar/Park-Dashboard/pull/13/files#diff-110bc809b1b51a37356fc9ef0586e5e43f81888620da72233e2232c0a8e90355

You can probably replace the second col with your component.

image

File: park-dashboard/src/Pages/Generic Charts/GenericCharts.js (hunk 2 above)
Comment (juhiaugustine) [inline park-dashboard/src/Pages/Generic Charts/GenericCharts.js:43]: I noticed a crash in this component when I was tapping on `Park Closure`. Can you fix this?

Steps to reproduce
- Go to Generic charts
- Scroll down to `AlertsVsParks` component
- Tap on `Park Closure`

image

**Review by juhiaugustine — CHANGES_REQUESTED** [review]

File: park-dashboard/src/Pages/NationalParkCharts/NationalParkChartsPage.js [hunk 4]
```diff
@@ -3,27 +3,36 @@ import { Container, Row, Col } from "react-bootstrap";
...
-          </h2>
-          <ActivitiesVsNP className="py-5 activities" />
+          <Col>
+            <h2 style={{ fontSize: "18px" }} className="text-center">
+              Donut Chart: Activities vs National Parks
+            </h2>
+            <ActivitiesVsNP className="py-5 activities" />
+          </Col>
         </Row>
+        <Col>
```
Comment (juhiaugustine) [inline park-dashboard/src/Pages/NationalParkCharts/NationalParkChartsPage.js]: I remember adding some code to make the charts more responsive? Can you pull the latest main branch and then modify your code here? 

This is my PR where I added code to make the charts responsive -> https://github.com/rutujapadgilwar/Park-Dashboard/pull/13/files#diff-110bc809b1b51a37356fc9ef0586e5e43f81888620da72233e2232c0a8e90355

You can probably replace the second col with your component.

image

**Review by juhiaugustine — CHANGES_REQUESTED** [review]

**Comment by schoure98** [comment]
As Juhi suggested in the last PR, can you change link colors to black? To maintain uniformaty throughout the website. Also if there's a different color for on:hover, it would look great.

**Comment by schoure98** [comment]
Screenshot 2023-06-04 at 11 50 25 PM

The Danger category is not representing any %? Is it possible to modify that?

I am getting the error "parkName is null" for Park Closure and Caution in generic charts.

File: park-dashboard/src/Pages/Generic Charts/GenericCharts.js (hunk 2 above)
Comment (rutujapadgilwar) [reply park-dashboard/src/Pages/Generic Charts/GenericCharts.js:43]: Hi @juhiaugustine, thank you for bringing this to my attention. I have addressed the issue and added null value validation to ensure proper functionality.

File: park-dashboard/src/Pages/NationalParkCharts/NationalParkChartsPage.js (hunk 4 above)
Comment (rutujapadgilwar) [reply park-dashboard/src/Pages/NationalParkCharts/NationalParkChartsPage.js]: I have implemented the responsiveness code here. Please review and validate it. Feel free to let me know if any further changes are required

**Comment by rutujapadgilwar** [comment]
> As Juhi suggested in the last PR, can you change link colors to black? To maintain uniformaty throughout the website. Also if there's a different color for on:hover, it would look great.

Hello @schoure98, as mentioned by Juhi for the last modal, I have included the same color properties for the links. They appear in blue and change to black when hovered over

**Comment by rutujapadgilwar** [comment]
> Screenshot 2023-06-04 at 11 50 25 PM
> The Danger category is not representing any %? Is it possible to modify that?
> 
> I am getting the error "parkName is null" for Park Closure and Caution in generic charts.

The issue is resolved now, I have added null value validation to ensure proper functionality.

**Review by juhiaugustine — APPROVED** [review]

**Pull request merged**